packages/owl-bot/src/copy-code.ts

```diff
@@ -663,7 +664,20 @@ export async function regeneratePullRequest(
 
   const sourceRepoCommitHash = copyTags[0].h;
 
-  cmd(`git checkout -b ${destBranch}`, {cwd: destDir});
+  if (action === 'regenerate') {
```

Comment:
Should both line 667 and 732 be action === regenerate? If so, why is there a type for 'append' (what does its code path look like)? More generally, what determines whether the action is regenerate or append, since there's only one label (I'm assuming)?

Reply:
> Should both line 667 and 732 be action === regenerate?

Yes.

> If so, why is there a type for 'append' (what does its code path look like)?  More generally, what determines whether the action is regenerate or append, since there's only one label (I'm assuming)?

There are two code paths through `copyCodeIntoPullRequest()`.

The first path is the path that existed before this PR, when the user checks the "Regenerate this pull request checkbox."  `copyCodeIntoPullRequest(action==="regenerate")` should behave exactly as `regeneratePullRequest()` did before this PR.  The tests were very useful to confirm this.

The new code path is for when Bootstrapper applies the label `owlbot:copy-code`.  In that case,  Owl Bot invokes `copyCodeIntoPullRequest(action==="append")`.  There are two important differences in the path:
1.  Instead copying the code into a copy of the main branch, Owl Bot copies the code into the PR's branch.
2.  Instead of rewriting the PR's title and body, it leaves them unchanged.